core/Updates/3.5.1-b1.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Updates;
+
+use Piwik\Filesystem;
+use Piwik\Notification;
+use Piwik\Plugin\Manager;
+use Piwik\SettingsServer;
+use Piwik\Updater;
+use Piwik\Updates as PiwikUpdates;
+
+class Updates_3_5_1_b1 extends PiwikUpdates
+{
+    public function doUpdate(Updater $updater)
+    {
+        // try to deactivate and remove the GeoIP2 plugin from marketplace,
+        // which causes problems with GeoIp2 plugin included in core
+        // Skip if new file `isoRegionNames.php` is detected within `GeoIP2` directory, as that means filesystem is case
+        // insensitive and GeoIP2 plugin has been partially overwritten by GeoIp2 plugin
+        try {
+            if (is_dir(PIWIK_INCLUDE_PATH . '/plugins/GeoIP2') && !file_exists(PIWIK_INCLUDE_PATH . '/plugins/GeoIP2/data/isoRegionNames.php')) {
```

Comment:
maybe we would need to add a test that makes sure this file exists? Just to be safe... if we otherwise ever remove the file, the plugin will be removed when someone updates from say an older Matomo version to a much newer. Be good to name the test clear to indicate what happens if the file does not exist.

Reply:
good point. Will add the test now